website/static/js/nodesPrivacy.js

```diff
@@ -165,114 +182,185 @@ var NodesPrivacyViewModel = function(parentIsPublic) {
+            }
+            else {
+                this.nodesChangedPrivate().push(nodesState[node].title);
             }
         }
-        self.nodesState(nodesState);
-        m.redraw(true);
-    };
+    }
+    this.page(this.CONFIRM);
+};
 
-    self.selectNone = function() {
-        var nodesState = ko.toJS(self.nodesState());
-        for (var node in nodesState) {
-            if (nodesState[node].canWrite) {
-                nodesState[node].public = false;
-                nodesState[node].changed = nodesState[node].public !== self.nodesOriginal[node].public;
+NodesPrivacyViewModel.prototype.confirmChanges =  function() {
+    var self = this;
 
-            }
+    var nodesState = ko.toJS(this.nodesState());
+    nodesState = Object.keys(nodesState).map(function(key) {
+        return nodesState[key];
+    });
+    var nodesChanged = nodesState.filter(function(node) {
+        return node.changed;
+    });
+    //The API's bulk limit is 100 nodes.  We catch the exception in nodes_privacy.mako.
+    if (nodesChanged.length <= 100) {
+        $osf.block('Updating Privacy');
+        patchNodesPrivacy(nodesChanged).then(function () {
+            self.onSetPrivacy(nodesChanged);
+
+            $osf.unblock();
+            self.nodesChangedPublic([]);
+            self.nodesChangedPrivate([]);
+            self.page(self.WARNING);
+            window.location.reload();
+        }).fail(function () {
+            $osf.unblock();
+            $osf.growl('Error', 'Unable to update project privacy');
+            Raven.captureMessage('Could not PATCH project settings.');
+            self.clear();
+            window.location.reload();
+        });
+    }
+};
+
+NodesPrivacyViewModel.prototype.clear = function() {
+    this.nodesChangedPublic([]);
+    this.nodesChangedPrivate([]);
+    this.page(this.WARNING);
+};
+
+NodesPrivacyViewModel.prototype.selectAll = function() {
+    var nodesState = ko.toJS(this.nodesState());
+    for (var node in nodesState) {
+        if (nodesState[node].canWrite) {
+            nodesState[node].public = true;
+            nodesState[node].changed = nodesState[node].public !== this.nodesOriginal[node].public;
         }
-        self.nodesState(nodesState);
-        m.redraw(true);
-    };
+    }
+    this.nodesState(nodesState);
+    m.redraw(true);
+};
 
-    self.back = function() {
-        var self = this;
-        self.nodesChangedPublic([]);
-        self.nodesChangedPrivate([]);
-        self.page(self.SELECT);
-    };
+NodesPrivacyViewModel.prototype.selectNone = function() {
+    var nodesState = ko.toJS(this.nodesState());
+    for (var node in nodesState) {
+        if (nodesState[node].canWrite) {
+            nodesState[node].public = false;
+            nodesState[node].changed = nodesState[node].public !== this.nodesOriginal[node].public;
+
+        }
+    }
+    this.nodesState(nodesState);
+    m.redraw(true);
+};
+
+NodesPrivacyViewModel.prototype.back = function() {
```

Comment:
+0 on moving all methods to the prototype. The memory savings we get from assigning to `prototype` is close to nil, since we almost always instantiate a ViewModel only once. Probably not even worth the extra typing (though I suppose we could use `oop.defclass` to reduce boilerplate), but this is fine as is.